Projects/AccountSDKIOSWeb/Tests/AccountSDKIOSWebTests/SimplifiedLogin/SharedKeychainSessionStorageFactoryTests.swift

```diff
@@ -0,0 +1,97 @@
+        
+        XCTAssertEqual(keychain.accessGroup, nil)
+    }
+    
+    func testReturnsAppKeychainWhenEntitlementIsMissing() {
+        let keychain = SharedKeychainSessionStorageFactory().makeKeychain(clientId: "client_id", service: "service_name", accessGroup: accessGroup, appIdentifierPrefix: "AZWSDFGHIJ")
+        
+        XCTAssertEqual(keychain.accessGroup, accessGroup)
+    }
+    
+    func testReturnsSharedKeychainWhenEntitlementExist() {
+        let keychainSessionStorageMock = MockKeychainSessionStorage(service: "service_name", accessGroup: sharedAccessGroup)
+        stub(keychainSessionStorageMock) { mock in
+            when(mock.checkEntitlements()).then { _ in
+                return Data()
+            }
+            when(mock.getAll()).then { _ in
+                return []
+            }
+        }
+        
+        let keychain = SharedKeychainSessionStorageFactory(keychain: nil, sharedKeychain: keychainSessionStorageMock).makeKeychain(clientId: "client_id", service: "service_name", accessGroup: accessGroup, appIdentifierPrefix: "AZWSDFGHIJ")
+        
+        XCTAssertIdentical(keychain, keychainSessionStorageMock)
+        verify(keychainSessionStorageMock).checkEntitlements()
+    }
+    
+    func testReturnsSharedKeychainWithoutMigrationWhenNonEmpty() {
+        let userSession = UserSession(clientId: "client_id", userTokens: Fixtures.userTokens, updatedAt: Date())
+        let sharedKeychainSessionStorageMock = MockKeychainSessionStorage(service: "service_name", accessGroup: sharedAccessGroup)
+        stub(sharedKeychainSessionStorageMock) { mock in
+            when(mock.checkEntitlements()).then { _ in
+                return Data()
+            }
+            when(mock.getAll()).then { _ in
+                return [userSession]
+            }
+        }
+        let keychainSessionStorageMock = MockKeychainSessionStorage(service: "service_name", accessGroup: accessGroup)
+        stub(keychainSessionStorageMock) { mock in
+            when(mock.get(forClientId: "client_id", completion: anyClosure()))
+                .then { _, completion in
+                    completion(userSession)
+                }
+        }
+        
+        let keychain = SharedKeychainSessionStorageFactory(keychain: keychainSessionStorageMock, sharedKeychain: sharedKeychainSessionStorageMock).makeKeychain(clientId: "client_id", service: "service_name", accessGroup: accessGroup, appIdentifierPrefix: "AZWSDFGHIJ")
+        
+        XCTAssertIdentical(keychain, sharedKeychainSessionStorageMock)
+        verify(sharedKeychainSessionStorageMock).checkEntitlements()
+        verify(sharedKeychainSessionStorageMock).getAll()
+        verify(sharedKeychainSessionStorageMock, never()).store(any(), accessGroup: sharedAccessGroup, completion: anyClosure())
+    }
+    
+    func testReturnsSharedKeychainWithUpdatedAccessGroupForItem() {
```

Comment:
The name of this function makes me think that this unit test is for assuring that the SharedKeychain gets its accessGroup updated.

But on diving in to the code a bit I realise the test is to assure that the sharedKeychain overwrite pre-existing userSession

Reply:
No, its not overwriting. This tests sceanrio:
- application has user session in standard keychain
- then app identifier is provided and entitlement is added to the app 
- so on next start i want to use shared keychain in my app but i also want to have my user session - so i migrate it 

this is why i have two mocks of keychain here

Reply:
Aha I see 👌 

So its testing migration to SharedKeychain